dipy/viz/ui.py

```diff
@@ -2952,6 +2952,278 @@ def set_img(self, img):
+                                 read_viz_icons(fname="stop2.png")))
+        self.button_icons.append(('checked',
+                                 read_viz_icons(fname="checkmark.png")))
+        self.button = Button2D(icon_fnames=self.button_icons,
+                               size=self.button_size)
+
+        self.text = TextBlock2D(text=self.label, font_size=self.font_size)
+
+    def _get_actors(self):
+        """ Get the actors composing this UI component.
+        """
+        return self.button.actors + self.text.actors
+
+    def _add_to_renderer(self, ren):
+        """ Add all subcomponents or VTK props that compose this UI component.
+
+        Parameters
+        ----------
+        ren : renderer
+        """
+        self.button.add_to_renderer(ren)
+        self.text.add_to_renderer(ren)
+
+    def _get_size(self):
+        width = self.button.size[0] + 10 + self.text.size[0]
+        height = max(self.button.size[1], self.text.size[1])
+        return np.array([width, height])
+
+    def _set_position(self, coords):
+        """ Position the lower-left corner of this UI component.
+
+        Parameters
+        ----------
+        coords: (float, float)
+            Absolute pixel coordinates (x, y).
+        """
+        num_newlines = self.label.count('\n')
+        self.button.position = coords + \
+            (0, num_newlines * self.font_size * 0.5)
+        offset = (self.button.size[0] + 10, 0)
+        self.text.position = coords + offset
+
+
+class Checkbox(UI):
+
+    """ A 2D set of :class:'Option' objects.
+    Multiple options can be selected.
+
+    Attributes
+    ----------
+    num_options : int
+        Number of options
+    labels : list(string)
+        List of labels of each option.
+    options : list(Option)
+        List of all the options in the checkbox set.
+    padding : float
+        Distance between two adjacent options
+    """
+
+    def __init__(self, labels, padding=1, font_size=18,
+                 font_family='Arial', position=(0, 0)):
+        """
+        Parameters
+        ----------
+        labels : list(string)
+            List of labels of each option.
+        padding : float
+            The distance between two adjacent options
+        font_size : int
+            Size of the text font.
+        font_family : str
+            Currently only supports Arial.
+        position : (float, float)
+            Absolute coordinates (x, y) of the lower-left corner of
+            the button of the first option.
+        """
+        self.labels = labels
+        self.num_options = len(labels)
+        self._padding = padding
+        self._font_size = font_size
+        self.font_family = font_family
+        super(Checkbox, self).__init__(position)
+
+    def _setup(self):
+        """ Setup this UI component.
+        """
+        self.options = []
+        button_y = self.position[1]
+        for option_no in range(self.num_options):
```

Comment:
Is `num_options` necessary in this class? It seems like you could write this loop as

```python
for label in self.labels
```

Otherwise, it is only used in `_get_size()`